src/loader.ts

```diff
@@ -223,6 +211,25 @@ export async function loadConfig<
   return r;
 }
 
+export async function resolveConfigPath<
+  T extends UserInputConfig = UserInputConfig,
+  MT extends ConfigLayerMeta = ConfigLayerMeta,
+>(options: LoadConfigOptions<T, MT>): Promise<string | undefined> {
+  normalizeLoadOptions(options);
+
+  // Load dotenv
+  if (options.dotenv) {
+    await setupDotenv({
```

Comment:
Do we really need this? It might be helpful for cloning private layers i guess during resoluition but both are hidden side-effects specially if users pass same configuration they expect to pass to resolver and assume it will only try to resolve.

I am feeling we could make it fully opt-in with something like `{ sideEffect: true }` or `{ loadLayers: true }`.